Makefile

```diff
@@ -67,39 +67,39 @@ release-controllers: update_crt_crd
 	./hack/generators/release-controllers/generate-release-controllers.py .
 
 checkconfig:
-	$(CONTAINER_ENGINE) run $(USER) --platform linux/${GO_ARCH} --rm -v "$(CURDIR):/release$(VOLUME_MOUNT_FLAGS)" gcr.io/k8s-prow/checkconfig:v20230901-e9e5d470a5 --config-path /release/core-services/prow/02_config/_config.yaml --supplemental-prow-config-dir=/release/core-services/prow/02_config --job-config-path /release/ci-operator/jobs/ --plugin-config /release/core-services/prow/02_config/_plugins.yaml --supplemental-plugin-config-dir /release/core-services/prow/02_config --strict --exclude-warning long-job-names --exclude-warning mismatched-tide-lenient
+	$(CONTAINER_ENGINE) run $(CONTAINER_ENGINE_OPTS) $(USER) --rm -v "$(CURDIR):/release$(VOLUME_MOUNT_FLAGS)" gcr.io/k8s-prow/checkconfig:v20230901-e9e5d470a5 --config-path /release/core-services/prow/02_config/_config.yaml --supplemental-prow-config-dir=/release/core-services/prow/02_config --job-config-path /release/ci-operator/jobs/ --plugin-config /release/core-services/prow/02_config/_plugins.yaml --supplemental-plugin-config-dir /release/core-services/prow/02_config --strict --exclude-warning long-job-names --exclude-warning mismatched-tide-lenient
```

Comment:
I might get a bit why GO_ARCH should be renamed (we could just call it `platform`). But I do not get the rest of the PR. 
```
$(CONTAINER_ENGINE) run $(CONTAINER_ENGINE_OPTS) $(USER) ...
```
From reading this line, I feel `$(USER)` is not an docker option but it actually is. Why do we have to have CONTAINER_ENGINE_OPTS for each cmd?

Reply:
The `CONTAINER_ENGINE_OPTS` is for user to override the execution parameters.

Reply:
This PR would enforce that each docker/podman cmd has to use THE SAME CONTAINER_ENGINE_OPTS.
We would lose the flexibility.
Do we want that?